src/lib/eina/eina_thread.c

```diff
@@ -228,30 +122,42 @@ EAPI Eina_Bool
 eina_thread_cancel(Eina_Thread t)
 {
    if (!t) return EINA_FALSE;
-   return pthread_cancel((pthread_t)t) == 0;
+   return _eina_thread_cancel(t);
 }
 
 EAPI Eina_Bool
 eina_thread_cancellable_set(Eina_Bool cancellable, Eina_Bool *was_cancellable)
 {
-   int state = cancellable ? PTHREAD_CANCEL_ENABLE : PTHREAD_CANCEL_DISABLE;
+   int state = cancellable ? EINA_THREAD_CANCEL_ENABLE : EINA_THREAD_CANCEL_DISABLE;
    int old = 0;
    int r;
-
+   #ifndef _WIN32
    /* enforce deferred in case users changed to asynchronous themselves */
-   pthread_setcanceltype(PTHREAD_CANCEL_DEFERRED, &old);
+   
+   pthread_setcanceltype(EINA_THREAD_CANCEL_DEFERRED, &old);
 
    r = pthread_setcancelstate(state, &old);
    if (was_cancellable && r == 0)
-     *was_cancellable = (old == PTHREAD_CANCEL_ENABLE);
-
+     *was_cancellable = (old == EINA_THREAD_CANCEL_ENABLE);
    return r == 0;
+   #else
+      if(!state){
+         *was_cancellable = EINA_TRUE;
+         return EINA_FALSE;
+      }else{
+         *was_cancellable = EINA_TRUE;
+         return EINA_TRUE;
+      }
+   #endif
 }
```

Comment:
Wouldn't it be better to implement this function at posix/win32 versions of eina_thread?

Reply:
This function was divided to _win32 and posix version. In windows version, it was implemented following Felipe's sugestions, because we haven't `pthread_setcanceltype` and `pthread_setcanceltype`. Furthermore you will find these functions in _win32.c, but it will go to @JPTIZ stubs.